subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -119,6 +119,54 @@ static inline void set_pkt_ll_addr(struct net_linkaddr *addr, bool comp,
 	addr->type = NET_LINK_IEEE802154;
 }
 
+/**
+ * Filters the destination address of the frame (used when IEEE802154_HW_FILTER
+ * is not available).
+ */
+static bool ieeee802154_check_dst_addr(struct net_if *iface,
+				      struct ieee802154_mhr *mhr)
+{
+	struct ieee802154_context *ctx = net_if_l2_data(iface);
+	const struct ieee802154_radio_api *radio =
+		net_if_get_device(iface)->api;
+
+	if (radio && (radio->get_capabilities(net_if_get_device(iface)) &
+		      IEEE802154_HW_FILTER)) {
+		return true;
+	}
+
+	if (mhr->fs->fc.dst_addr_mode == IEEE802154_ADDR_MODE_NONE) {
+		return true;
+
+	} else if (mhr->fs->fc.dst_addr_mode == IEEE802154_ADDR_MODE_EXTENDED) {
+		if (mhr->dst_addr->plain.pan_id == 0xFFFF) {
```

Comment:
I think you should still validate the destination address, even for broadcast PAN id. And you should likely also filter out frames that have a different PAN id value than ours (there's `pan_id` field in the `ctx`). @hubertmis Is that correct?

Reply:
Long story short: that's correct. You need to check destination address even if PAN Id is broadcast.

And now the details:

From IEEE 802.15.4 Specification:

> If any of the fourth-level filtering requirements are not satisfied, the MAC sublayer shall discard the
incoming frame without processing it further.

One of fourth-level filtering rules is verifying PAN Id, another one is verifying destination address.